src/infrastructure/validation/testing/ITest.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * @author - Abhishek Saran(111701001)
+ * @module - Infrastucture module 
+ * @team - Test Harness
+ * @description - ITest interface with abstract class TestCase 
+ * @summary - Modules have to extend the abstract class to implement the run method for each test case  
+ */
+
+
+interface ITest{
+
+	//Methods
+	void setDescription(String des);
+	String getDescription();
+	void setError(String err);
+	String getError();
+	void setCategory();
+	String getCategory(String c);
+	void setPriority(int p);
+	String getPriority();
+
+	boolean run();
+
+}
+
+public abstract class TestCase implements ITest {
+
+	//data members
+	//category of a test case as its module name 
+	private String category;
+	
+	//error occured in the test case 
+  private String error;
+  
+  //a small description about what this test case is about
+  private String description;
+  
+  //the priority level of the test case - 0 being lowest and 2 being highest
+  private int priority;
```

Comment:
Write all the data members together. Write one block of comments above these variables using @param to describe each data member.